### PR TITLE
Release 8.0.1 - add dependency constraint for omniauth 1.x series.

### PR DESCRIPTION
**Pull request by koshilife**
Hi,@simi

Could you check this about #357.

references

- dependency for omniauth-oauth v1.2
  - https://rubygems.org/gems/omniauth-oauth2/versions/1.2.0
- referred to other strategy PR
  - https://github.com/omniauth/omniauth-github/pull/95

### CODE REVIEW

File: omniauth-facebook.gemspec [hunk 1]
```diff
@@ -16,7 +16,8 @@ Gem::Specification.new do |s|
   s.executables   = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
   s.require_paths = ['lib']
 
-  s.add_runtime_dependency 'omniauth-oauth2', '~> 1.2'
+  s.add_runtime_dependency 'omniauth', '~> 1.2'
+  s.add_runtime_dependency 'omniauth-oauth2', '>= 1.2', '< 1.7.1'
```
Comment (swiknaba) [inline omniauth-facebook.gemspec:20]: why do you want to hard-lock this to `< 1.7.1`? I am running this with omniauth-oauth2 in version `1.7.1` just fine. This change breaks installations (if this is going through, than this is **not** a minor change, but a breaking one).

Comment (koshilife) [reply omniauth-facebook.gemspec:20]: @swiknaba thanks quick review and help.
I was misunderstanding that omniauth-oauth2 version 1.7.1 does not support omniauth1.x.
You are right and this commit is wrong, i'm really sorry.

Could you advise for fix?
How about below dependency?

## v8.0.1: support OmniAuth 1.x

```
  s.add_runtime_dependency 'omniauth', '~> 1.2'
  s.add_runtime_dependency 'omniauth-oauth2', '>= 1.2'
```

## v9.0.0: support OmniAuth 2.0 series

```
  s.add_runtime_dependency 'omniauth', '~> 2.0'
  s.add_runtime_dependency 'omniauth-oauth2', '~> 1.7.1'
```

Comment (koshilife) [reply omniauth-facebook.gemspec]: thanks!

I undid 'omniauth-oauth2' dependency, so this changs mean to add `'omniauth', '~> 1.2'` only.

Comment (swiknaba) [reply omniauth-facebook.gemspec]: I just checked, we did actually the same dependency updates as you in your two PRs in a private gem for an internal OAuth implementation. We're running that in production for a while now.

Comment (koshilife) [reply omniauth-facebook.gemspec]: Thank you for checking, let me know when if you need to fix something anytime.

**Pull request closed without merging**